### ISSUE TITLE
Bibtex citation completion requires full path of bib file

### ISSUE THREAD

**Issue by Shrimpram** [lervag/vimtex#2184]
### Description

When trying to complete citations with bibtex, I was running into errors when I simply included the file name of the bib file. My bib files are located in my texmf tree directory, and are found by kpsewhich. However, in order for citation completions to work in vimtex I need to use the full path of the bib file.

E.g. merely having `\addbibresource{somefile.bib}` does not make the citation completion menu appear when I press <kbd>\<C-x\>\<C-o\></kbd>, but I need to include the whole path, `\addbibresource{~/Library/texmf/bibtex/bib/somefile.bib}`.

### Steps to reproduce

1. `nvim -u minimal.vim minimal.tex`
2. Uncomment out the first bib file option "`\addbibresource{somefile.bib}`"
3. Go over to `\cite{}`
4. Try to complete citations using <kbd>\<C-x\>\<C-o\></kbd>

<details>
<summary>Here is the `minimal.vim` file</summary>

```vim
set nocompatible
let &runtimepath  = '~/test/vimtex,' . &runtimepath
let &runtimepath .= ',~/test/vimtex/after'
filetype plugin indent on
syntax enable
" Add relevant options and VimTeX configuration below.
```

</details>

<details>
<summary>Here is the `minimal.tex`</summary>

```tex
\documentclass{minimal}
\usepackage{biblatex}

% \addbibresource{somefile.bib}
% \addbibresource{~/Library/texmf/bibtex/bib/somefile.bib}

\begin{document}

\cite{}

\end{document}
```
</details>

### Expected behavior

The citation completion menu should pop up with all of the bibliography entries in `somefile.bib`

### Actual behavior

There is no citation completion when using `\addbibresource{somefile.bib}`, but there *is* citation completion when using `\addbibresource{~/Library/texmf/bibtex/bib/somefile.bib}`.

However, compilation works on both versions without error and `kpsewhich somefile.bib` returns the location, the only issue is in completion.

### Do you use a latexmkrc file?

No

### VimtexInfo

```yaml
System info
  OS: macOS 11.6 (20G165)
  Vim version: NVIM v0.6.0-dev+356-gbec7f47ce
  Has clientserver: true
  Servername: /var/folders/zh/j15t9snn5wdbyz3rh4gxj6t00000gn/T/nvimVMydBY/0

VimTeX project: minimal
  base: minimal.tex
  root: /Users/shreeram/test
  tex: /Users/shreeram/test/minimal.tex
  main parser: current file verified
  document class: minimal
  packages: bibtex
  compiler: latexmk
    engine: -xelatex
    options:
      -verbose
      -file-line-error
      -synctex=1
      -interaction=nonstopmode
    callback: 1
    continuous: 1
    executable: latexmk
  viewer: General
  qf method: LaTeX logfile
```

**Comment by clason** [comment]
`\addbibresource` is from Bib**La**TeX, not BibTeX, and VimTeX checks loaded packages for enabling features.

TL;DR: your `minimal.tex` has a typo; it's `\usepackage{biblatex}`, not `\usepackage{bibtex}`.

**Comment by Shrimpram** [comment]
Apologies, I fixed that typo but the issue is still there.

**Comment by lervag** [comment]
First: I've fixed the typo in your original post as well.

Now, I'm curious if this could be an issue with the caching mechanism. Can you try to clear the cache with `:VimtexClearCache kpsewhich` and then try again with the relative path?

**Comment by Shrimpram** [comment]
Yes clearing the cache solved the issue, the relative path opens the citation completion menu. Thank you for your help!

**Comment by lervag** [comment]
Great, happy to hear it. However, you actually found a bug here. I've fixed it, but after updating, you will need to clear the cache again.

**Comment by lervag** [comment]
Just to add to this for potential interested readers, including future self: It seems more or less common to use the same name for reference file in different projects, e.g. `reference.bib`. The old implementation did not account for the fact that a file like `reference.bib` may be different in different projects. That is, since the results were cached, every project after the first would resolve the `reference.bib` that was local to the first project, and not the one local to itself. This also means that one could potentially have strange completion issues because the resolved bib file was not the correct one. In fact, I am sure this must have happened to a lot of people, so I'm surprised it has not been reported before!
